backend/src/db/mod.rs

```diff
@@ -61,27 +116,76 @@ pub(crate) async fn create_pool(config: &DbConfig) -> Result<Pool> {
         host: Some(config.host.clone()),
         port: Some(config.port),
         dbname: Some(config.database.clone()),
+        ssl_mode: Some(if config.tls_mode == TlsMode::Off {
+            deadpool_postgres::SslMode::Disable
+        } else {
+            deadpool_postgres::SslMode::Require
+        }),
         .. PoolConfig::default()
     };
 
     debug!(
-        "Connecting to postgresql://{}:*****@{}:{}/{}",
+        "Connecting to 'postgresql://{}:*****@{}:{}/{}' (TLS: {:?})",
         config.user,
         config.host,
         config.port,
         config.database,
+        config.tls_mode,
     );
 
-    let pool = pool_config.create_pool(Some(Runtime::Tokio1), NoTls)?;
+    // Handle TLS and create pool.
+    let pool = if config.tls_mode == TlsMode::Off {
+        pool_config.create_pool(Some(Runtime::Tokio1), NoTls)?
+    } else {
+        // Prepare certificate store. If we do not verify the certificate, it's
+        // just empty. Otherwise we load system-wide root CAs.
+        let mut root_certs = rustls::RootCertStore::empty();
+        if config.tls_mode == TlsMode::On {
+            let system_certs = rustls_native_certs::load_native_certs()
+                .context("failed to load all system-wide certificates")?;
+
+            let count = system_certs.len();
+            for cert in system_certs {
+                root_certs.add(&rustls::Certificate(cert.0))
+                    .context("failed to load system-wide certificate")?;
+            }
+            debug!("Loaded {count} system-wide certificates");
+
+            if let Some(cert_path) = &config.server_cert {
+                load_pem_file(cert_path, &mut root_certs)
+                    .with_context(|| format!("failed to load '{}'", cert_path.display()))?;
+                debug!(
+                    "Loaded {} certificates from '{}'",
+                    root_certs.len() - count,
```

Comment:
This line confused me, which mostly wasn't your fault, but still: Maybe `count` is not the best name for this variable. Maybe something like `system_count`? :thinking: :shrug:

Reply:
Alternatively, or in addition to that, you could have `load_pem_file` return the number of certs it read, instead of calculating the number here. Then you don't need `count` here, anymore.